conversation/frontend/src/hooks/useMessageAttachments.tsx

```diff
@@ -67,44 +67,57 @@ export function useMessageAttachments({ id }: Message) {
     return Promise.all(attachments.map(({ id }) => detachFile(id)));
   }
 
+  /**
+   * Copy files to workspace
+   * @param attachments - The attachments to copy
+   * @param selectedFolderId - The folder ID to copy the files to
+   * @returns {Promise<boolean>} - Returns true if the operation was successful, false otherwise
+   */
   async function copyToWorkspace(
     attachments: Attachment[],
     selectedFolderId: string,
   ) {
-    try {
-      const downloadFilesPromises = attachments.map(async (attachment) => {
-        const attachmentBlob = await downloadAttachmentMutation.mutateAsync({
-          messageId: id,
-          attachmentId: attachment.id,
-        });
+    const files = await downloadAttachements(attachments);
+    return sendFilesToWorkspace(files, selectedFolderId);
+  }
 
-        if (!attachmentBlob) return;
+  async function downloadAttachements(attachments: Attachment[]) {
+    const downloadFilesPromises = attachments.map(async (attachment) => {
+      const attachmentBlob = await downloadAttachmentMutation.mutateAsync({
+        messageId: id,
+        attachmentId: attachment.id,
+      });
 
-        return new File([attachmentBlob], attachment.filename, {
-          type: attachment.contentType,
-        });
+      if (!attachmentBlob) return;
+
+      return new File([attachmentBlob], attachment.filename, {
+        type: attachment.contentType,
       });
+    });
 
-      const files = await Promise.all(downloadFilesPromises);
+    const files = await Promise.all(downloadFilesPromises);
+    return files.filter((file) => !!file);
+  }
 
-      // 2 - Send files to workspace
+  async function sendFilesToWorkspace(files: File[], selectedFolderId: string) {
+    try {
       const addFilesPromises = files.map((file) => {
-        if (!file) return;
-
         return odeServices.workspace().saveFile(file, {
```

Comment:
Puisque la lambda fait maintenant un `return` directement, on peut simplifier l'écriture
```suggestion
await Promise.all( files.map((file) => odeServices.workspace().saveFile(file, {parentId: selectedFolderId})) );
toast.success(
     t('conversation.notify.copyToWorkspace', { count: files.length }),
);
```

Reply:
j'ai simplifié à moitié, je trouve ça plus facile à lire si découpé en deux partie

```suggestion
         const addFilesPromises = files.map((file) =>
        odeServices.workspace().saveFile(file, {
          parentId: selectedFolderId,
        }),
      );
      await Promise.all(addFilesPromises);
```

Reply:
Ok
Mais à la réflexion, d'un point de vue algorithmique, il y a un soucis.

* On fait X upload en parallèle => très bien
* Mais si 1 upload échoue, le front considère que tous ont échoué. 
Or c'est faux, certaines PJs seront bien rattachées du point de vue backend.

On devrait plutôt filtrer les uploads qui ont échoué, et afficher dans le toast le nombre qui a réussi et le nombre qui a échoué (faire 2 toasts ?)